subiquity/server/controllers/filesystem.py

```diff
@@ -558,21 +558,21 @@ def update_devices(self, device_map):
                 if action in self._device_to_structure:
                     self._device_to_structure[action].device = path
 
-    def guided_direct(self, gap):
+    async def guided_direct(self, gap):
         spec = dict(fstype="ext4", mount="/")
-        self.create_partition(device=gap.device, gap=gap, spec=spec)
+        await self.create_partition(device=gap.device, gap=gap, spec=spec)
 
     def guided_dd(self, disk: ModelDisk):
         self.model.dd_target = disk
 
-    def guided_lvm(self, gap, choice: GuidedChoiceV2):
+    async def guided_lvm(self, gap, choice: GuidedChoiceV2):
         device = gap.device
         part_align = device.alignment_data().part_align
         bootfs_size = align_up(sizes.get_bootfs_size(gap.size), part_align)
         gap_boot, gap_rest = gap.split(bootfs_size)
         spec = dict(fstype="ext4", mount="/boot")
-        self.create_partition(device, gap_boot, spec)
-        part = self.create_partition(device, gap_rest, dict(fstype=None))
+        await self.create_partition(device, gap_boot, spec)
```

Comment:
My intention was that at the end, the client would not use (and not inherit anymore from) the FilesystemManipulator. And if only the server uses the manipulator, then there should be no need for all the async stuff. `create_partition` would still be implemented in the manipulator but used by the server only.

But unless I can replace all the calls from the client to the manipulator with API calls in one go, I will run into problems with code that is not async.

I did consider a few ways to avoid making everything suddenly async but couldn't come up with something that would theoretically work.

I'm happy to try other solutions. It feels like the wrapper will not be enough to make it possible to call async code *from* methods of the manipulator ; which is what we'd need in practice AFAICT.